tools/scripts/utils/piece-script-utils.ts

```diff
@@ -61,19 +62,42 @@ export async function findAllPiecesDirectoryInSource(): Promise<string[]> {
     const paths = await traverseFolder(piecesPath)
     return paths
 }
+
+export async function findAllPremiumPiecesDirectory(): Promise<string[]> {
+    const piecesPath = resolve(cwd(), 'packages', 'ee', 'pieces')
+    const paths = await traverseFolder(piecesPath)
+    return paths
+}
+
 export async function findPieceDirectoryInSource(pieceName: string): Promise<string | null> {
-    const piecesPath =  await findAllPiecesDirectoryInSource()
+    const piecesPath = [...await findAllPiecesDirectoryInSource(), ...await findAllPremiumPiecesDirectory()]
     const piecePath = piecesPath.find((p) => p.includes(pieceName))
     return piecePath ?? null
 }
 
 export async function findAllPieces(): Promise<PieceMetadata[]> {
     const piecesPath = resolve(cwd(), 'dist', 'packages', 'pieces')
     const paths = await traverseFolder(piecesPath)
-    const pieces = await Promise.all(paths.map((p) => loadPieceFromFolder(p)))
+    const enterprisePiecesPaths = await buildAndFindAllPremiumPieces()
+    const pieces = await Promise.all([...paths ,...enterprisePiecesPaths].map((p) => loadPieceFromFolder(p)))
     return pieces.filter((p): p is PieceMetadata => p !== null).sort(byDisplayNameIgnoreCase)
 }
 
+async function buildAndFindAllPremiumPieces(): Promise<string[]> {
+    const piecesFolder = await findAllPremiumPiecesDirectory()
+    if (piecesFolder.length === 0) {
+        return []
+    }
+    for (const pieceFolder of piecesFolder) {
+        const projectJson = await readProjectJson(pieceFolder);
+        await exec('npx nx build ' + projectJson.name);
+    }
+
+    const enterprisePiecesPath = resolve(cwd(), 'dist', 'packages', 'ee', 'pieces')
```

Comment:
Can you restrict them by edition, you don't want people to see them in the dev mode and think it okay to use them